sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/LiteralSuffixUpperCase.cs

```diff
@@ -48,14 +49,17 @@ protected override void Initialize(SonarAnalysisContext context)
                     var literal = (LiteralExpressionSyntax)c.Node;
                     var text = literal.Token.Text;
 
-                    if (text[text.Length - 1] == 'l' &&
-                        c.SemanticModel.GetTypeInfo(literal).Type.Is(KnownType.System_Int64))
+                    if (text[text.Length - 1] == 'l' && NotUnsigned(text))
```

Comment:
We can use here [EndsWith(Char)
](https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=net-5.0#System_String_EndsWith_System_Char_)

Reply:
it was added in .NET Standard 2.1, and we're still on 2.0.

Reply:
Then maybe the [overload with string](https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=netframework-4.8#overloads)? I think it's easier to read.